<!DOCTYPE html><html lang="en">
<head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8">
<title>[1710.02431] Emerging whole-cell modeling principles and methods</title><meta property="og:description" content="Whole-cell computational models aim to predict cellular phenotypes from genotype by representing the entire genome, the structure and concentration of each molecular species, each molecular interaction, and the extrace…">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Emerging whole-cell modeling principles and methods">
<meta name="twitter:image:src" content="https://ar5iv.labs.arxiv.org/assets/ar5iv_card.png">
<meta name="twitter:image:alt" content="ar5iv logo">
<meta property="og:title" content="Emerging whole-cell modeling principles and methods">
<meta property="og:site_name" content="ar5iv">
<meta property="og:image" content="https://ar5iv.labs.arxiv.org/assets/ar5iv_card.png">
<meta property="og:type" content="article">
<meta property="og:url" content="https://ar5iv.labs.arxiv.org/html/1710.02431">
<link rel="canonical" target="_blank" href="https://ar5iv.labs.arxiv.org/html/1710.02431">

<!--Generated on Sun Mar  3 03:22:41 2024 by LaTeXML (version 0.8.8) http://dlmf.nist.gov/LaTeXML/.-->
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

<script>
  function detectColorScheme(){
    var theme="light";
    var current_theme = localStorage.getItem("ar5iv_theme");
    if(current_theme){
      if(current_theme == "dark"){
        theme = "dark";
      } }
    else if(!window.matchMedia) { return false; }
    else if(window.matchMedia("(prefers-color-scheme: dark)").matches) {
      theme = "dark"; }
    if (theme=="dark") {
      document.documentElement.setAttribute("data-theme", "dark");
    } else {
      document.documentElement.setAttribute("data-theme", "light"); } }

  detectColorScheme();

  function toggleColorScheme(){
    var current_theme = localStorage.getItem("ar5iv_theme");
    if (current_theme) {
      if (current_theme == "light") {
        localStorage.setItem("ar5iv_theme", "dark"); }
      else {
        localStorage.setItem("ar5iv_theme", "light"); } }
    else {
        localStorage.setItem("ar5iv_theme", "dark"); }
    detectColorScheme(); }
</script>
<link media="all" rel="stylesheet" href="/assets/ar5iv-fonts.0.8.0.min.css"><link media="all" rel="stylesheet" href="/assets/ar5iv.0.8.0.min.css"><link media="all" rel="stylesheet" href="/assets/ar5iv-site.0.2.2.css">
</head>
<body>
<div class="ltx_page_main">
<div class="ltx_page_content">
<article class="ltx_document ltx_fleqn">
<h1 class="ltx_title ltx_title_document">Emerging whole-cell modeling principles and methods</h1>
<div class="ltx_authors">
<span class="ltx_creator ltx_role_author">
<span class="ltx_personname">Arthur P. Goldberg
</span></span>
<span class="ltx_creator ltx_role_author">
<span class="ltx_personname">Balázs Szigeti
</span></span>
<span class="ltx_creator ltx_role_author">
<span class="ltx_personname">Yin Hoon Chew
</span></span>
<span class="ltx_creator ltx_role_author">
<span class="ltx_personname">John A. P. Sekar
</span></span>
<span class="ltx_creator ltx_role_author">
<span class="ltx_personname">Yosef D. Roth
</span></span>
<span class="ltx_creator ltx_role_author">
<span class="ltx_personname">Jonathan R. Karr
</span><span class="ltx_author_notes">
<span class="ltx_contact ltx_role_address">Icahn Institute for Genomics and Multiscale Biology
</span>
<span class="ltx_contact ltx_role_address">Department of Genetics and Genomic Sciences
<br class="ltx_break">Icahn School of Medicine at Mount Sinai, New York, New York, 10029, USA
</span>
<span class="ltx_contact ltx_role_address">These authors contributed equally to the work.
</span></span></span>
</div>

<div class="ltx_abstract">
<h6 class="ltx_title ltx_title_abstract">Abstract</h6>
<p id="id1.id1" class="ltx_p">Whole-cell computational models aim to predict cellular phenotypes from genotype by representing the entire genome, the structure and concentration of each molecular species, each molecular interaction, and the extracellular environment. Whole-cell models have great potential to transform bioscience, bioengineering, and medicine. However, numerous challenges remain to achieve whole-cell models. Nevertheless, researchers are beginning to leverage recent progress in measurement technology, bioinformatics, data sharing, rule-based modeling, and multi-algorithmic simulation to build the first whole-cell models. We anticipate that ongoing efforts to develop scalable whole-cell modeling tools will enable dramatically more comprehensive and more accurate models, including models of human cells.</p>
</div>
<span id="id1" class="ltx_note ltx_note_frontmatter ltx_role_journal"><sup class="ltx_note_mark">†</sup><span class="ltx_note_outer"><span class="ltx_note_content"><sup class="ltx_note_mark">†</sup><span class="ltx_note_type">journal: </span>Current Opinion in Biotechnology</span></span></span>
<section id="Sx1" class="ltx_section">
<h2 class="ltx_title ltx_font_bold ltx_title_section">Introduction</h2>

<div id="Sx1.p1" class="ltx_para ltx_noindent">
<p id="Sx1.p1.1" class="ltx_p">Whole-cell (WC) computational models aim to predict cellular phenotypes from genotype and the environment by representing the function of each gene, gene product, and metabolite <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib1" title="" class="ltx_ref">1</a>]</cite>. WC models could unify our understanding of cell biology and enable researchers to perform <span id="Sx1.p1.1.1" class="ltx_text ltx_font_italic">in silico</span> experiments with complete control, scope, and resolution <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib2" title="" class="ltx_ref">2</a>, <a href="#bib.bib3" title="" class="ltx_ref">3</a>]</cite>. WC models could also help bioengineers rationally design microorganisms that can produce useful chemicals and act as biosensors, and help physicians design personalized therapies tailored to each patient’s genome.</p>
</div>
<div id="Sx1.p2" class="ltx_para ltx_noindent">
<p id="Sx1.p2.1" class="ltx_p">Despite their potential, there is little consensus on how WC models should represent cells, what phenotypes WC models should predict, or how to achieve WC models. Nevertheless, we and others are beginning to leverage advances in measurement technology, bioinformatics, rule-based modeling, and multi-algorithmic simulation to develop WC models <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib4" title="" class="ltx_ref">4</a>, <a href="#bib.bib5" title="" class="ltx_ref">5</a>, <a href="#bib.bib6" title="" class="ltx_ref">6</a>, <a href="#bib.bib7" title="" class="ltx_ref">7</a>, <a href="#bib.bib8" title="" class="ltx_ref">8</a>, <a href="#bib.bib9" title="" class="ltx_ref">9</a>]</cite>. However, substantial work remains to achieve WC models <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib10" title="" class="ltx_ref">10</a>, <a href="#bib.bib11" title="" class="ltx_ref">11</a>]</cite>.</p>
</div>
<div id="Sx1.p3" class="ltx_para ltx_noindent">
<p id="Sx1.p3.1" class="ltx_p">To build consensus on WC modeling, we propose a set of key physical and chemical mechanisms that WC models should aim to represent, and a set of key phenotypes that WC models should aim to predict. We also summarize the experimental and computational progress that is making WC modeling feasible, and outline several technological advances that would help accelerate WC modeling.</p>
</div>
<div id="Sx1.p4" class="ltx_para ltx_noindent">
<p id="Sx1.p4.1" class="ltx_p">Note, our proposals focus on defining WC models that are needed for research studies and applications such as bioengineering and personalized medicine which depend on understanding the molecular details of the majority of intracellular processes. However, research that depends on fewer intracellular processes could be served by smaller, more focused models.</p>
</div>
</section>
<section id="Sx2" class="ltx_section">
<h2 class="ltx_title ltx_font_bold ltx_title_section">Physics and chemistry that WC models should aim to represent</h2>

<div id="Sx2.p1" class="ltx_para ltx_noindent">
<p id="Sx2.p1.1" class="ltx_p">We propose that WC models aim to represent all of the chemical reactions in a cell and all of the physical processes that influence their rates (Figure <a href="#Sx2.F1" title="Figure 1 ‣ Physics and chemistry that WC models should aim to represent ‣ Emerging whole-cell modeling principles and methods" class="ltx_ref"><span class="ltx_text ltx_ref_tag">1</span></a>a). This requires representing (a) the <span id="Sx2.p1.1.1" class="ltx_text ltx_font_bold">sequence</span> of each chromosome, RNA, and protein; the location of each chromosomal feature, including each gene, operon, promoter, and terminator; and the location of each site on each RNA and protein; (b) the <span id="Sx2.p1.1.2" class="ltx_text ltx_font_bold">structure</span> of each molecule, including atom-level information about small molecules, the domains and sites of macromolecules, and the subunit composition of complexes; (c) the <span id="Sx2.p1.1.3" class="ltx_text ltx_font_bold">subcellular organization</span> of cells into organelles and microdomains; (d) the participants and effect of each <span id="Sx2.p1.1.4" class="ltx_text ltx_font_bold">molecular interaction</span>, including the molecules that are consumed, produced, and transported, the molecular sites that are modified, and the bonds that are broken and formed, (e) the <span id="Sx2.p1.1.5" class="ltx_text ltx_font_bold">kinetic parameters</span> of each interaction; (f) the <span id="Sx2.p1.1.6" class="ltx_text ltx_font_bold">concentration</span> of each species in each organelle and microdomain; and (g) the concentration of each species in the <span id="Sx2.p1.1.7" class="ltx_text ltx_font_bold">extracellular environment</span>. In addition, to enable WC models to be rigorously tested, each WC model should represent a single, well-defined experimental system. To minimize the complexity of WC models, we recommend modeling small, fast-growing, non-adherent, autonomous, self-renewing cells growing on defined, rich, homogeneous media. Together, this would enable WC models to describe how cellular behavior emerges from the combined function of each gene and genetic variant, and capture how cells respond to changes in their internal and external environments.</p>
</div>
<figure id="Sx2.F1" class="ltx_figure"><img src="/html/1710.02431/assets/x1.png" id="Sx2.F1.g1" class="ltx_graphics ltx_centering ltx_img_landscape" width="461" height="149" alt="Refer to caption">
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">Figure 1: </span> The physical and chemical mechanisms that WC models should aim to represent (a) and the phenotypes that WC models should aim to predict (b).</figcaption>
</figure>
</section>
<section id="Sx3" class="ltx_section">
<h2 class="ltx_title ltx_font_bold ltx_title_section">Phenotypes that WC models should aim to predict</h2>

<div id="Sx3.p1" class="ltx_para ltx_noindent">
<p id="Sx3.p1.1" class="ltx_p">We also propose that WC models aim to predict the behavioral trajectories of single cells over their life cycles, with each simulation representing a different cell within a heterogeneous clonal population (Figure <a href="#Sx2.F1" title="Figure 1 ‣ Physics and chemistry that WC models should aim to represent ‣ Emerging whole-cell modeling principles and methods" class="ltx_ref"><span class="ltx_text ltx_ref_tag">1</span></a>b). This should include behaviors within individual cells such as the <span id="Sx3.p1.1.1" class="ltx_text ltx_font_bold">stochastic dynamics</span> of each molecular interaction; the <span id="Sx3.p1.1.2" class="ltx_text ltx_font_bold">temporal dynamics</span> of the concentration of each species; the <span id="Sx3.p1.1.3" class="ltx_text ltx_font_bold">spatial dynamics</span> of the concentration of each species in each organelle and microdomain; and <span id="Sx3.p1.1.4" class="ltx_text ltx_font_bold">complex phenotypes</span> such as cell shape, growth rate, motility, and fate, as well as the <span id="Sx3.p1.1.5" class="ltx_text ltx_font_bold">variation</span> in the behavior of single cells within clonal populations. Together, this would enable WC models to capture how stochastic and single-cell variation can generate phenotypic diversity; how a cell responds to external cues such as nutrients, growth factors and drugs; and how a cell coordinates critical events such as the G1/S transition. This would also enable WC models to generate predictions that could be embedded into higher-order multiscale models. For example, WC models could predict the timing and speed of chemotaxis, which could help multiscale models predict tumor metastasis.</p>
</div>
</section>
<section id="Sx4" class="ltx_section">
<h2 class="ltx_title ltx_font_bold ltx_title_section">Available resources</h2>

<div id="Sx4.p1" class="ltx_para ltx_noindent">
<p id="Sx4.p1.1" class="ltx_p">Achieving WC models will require extensive data to constrain every parameter. Fortunately, measurement technology is rapidly advancing. Here, we review the latest methods for generating data for WC models, and highlight repositories and other resources that contain useful data for WC modeling.</p>
</div>
<section id="Sx4.SSx1" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Measurement methods</h3>

<div id="Sx4.SSx1.p1" class="ltx_para ltx_noindent">
<p id="Sx4.SSx1.p1.1" class="ltx_p">Advances in single-cell and genomic measurement are rapidly generating data that could be used for WC modeling <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib12" title="" class="ltx_ref">12</a>, <a href="#bib.bib13" title="" class="ltx_ref">13</a>, <a href="#bib.bib14" title="" class="ltx_ref">14</a>]</cite> (Table S1). For example, Meth-Seq can assess epigenetic modifications <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib15" title="" class="ltx_ref">15</a>]</cite>, Hi-C can determine chromosome structures <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib16" title="" class="ltx_ref">16</a>]</cite>, ChIP-seq can determine protein-DNA interactions <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib17" title="" class="ltx_ref">17</a>]</cite>, fluorescence microscopy can determine protein localizations, mass-spectrometry can quantitate metabolite and protein concentrations, FISH <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib18" title="" class="ltx_ref">18</a>]</cite> and scRNA-seq <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib19" title="" class="ltx_ref">19</a>]</cite> can quantitate the dynamics and single-cell variation of RNA abundances, and fluorescence microscopy and mass cytometry <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib20" title="" class="ltx_ref">20</a>]</cite> can quantitate the dynamics and single-cell variation of protein abundances. In particular, WC models can be constrained by combining high-dimensional measurement methods with multiple genetic and environmental perturbations, frequent temporal observations, and cutting-edge distributed parameter estimation methods. However, substantial work remains to develop methods that can measure non-model organisms including small, slow-growing, and unculturable cells.</p>
</div>
</section>
<section id="Sx4.SSx2" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Data repositories</h3>

<div id="Sx4.SSx2.p1" class="ltx_para ltx_noindent">
<p id="Sx4.SSx2.p1.1" class="ltx_p">Researchers are also rapidly aggregating much of the data needed for WC modeling into public repositories (Table S2). For example, UniProt contains a multitude of information about proteins <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib21" title="" class="ltx_ref">21</a>]</cite>; BioCyc contain extensive information about interactions <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib22" title="" class="ltx_ref">22</a>]</cite>; ECMDB <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib23" title="" class="ltx_ref">23</a>]</cite>, ArrayExpress <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib24" title="" class="ltx_ref">24</a>]</cite>, and PaxDb <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib25" title="" class="ltx_ref">25</a>]</cite> contain metabolite, RNA, and protein abundances, respectively; and SABIO-RK contains kinetic parameters <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib26" title="" class="ltx_ref">26</a>]</cite>. Furthermore, meta-databases such as <span id="Sx4.SSx2.p1.1.1" class="ltx_text ltx_font_italic">Nucleic Acid Research</span>’s Database Summary contain lists of repositories <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib27" title="" class="ltx_ref">27</a>]</cite>.</p>
</div>
</section>
<section id="Sx4.SSx3" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Prediction tools</h3>

<div id="Sx4.SSx3.p1" class="ltx_para ltx_noindent">
<p id="Sx4.SSx3.p1.1" class="ltx_p">For certain types of data, accurate prediction tools can be superior to direct experimental evidence which may have incomplete coverage or may be limited to a small number of genotypes and environments. Currently, many tools can predict properties such as operons, RNA folds, and protein localizations (Table S3). For example, PSORTb predicts the localization of bacterial proteins <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib28" title="" class="ltx_ref">28</a>]</cite>. However, many current prediction tools lack sufficient accuracy for WC modeling.</p>
</div>
</section>
<section id="Sx4.SSx4" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Published models</h3>

<div id="Sx4.SSx4.p1" class="ltx_para ltx_noindent">
<p id="Sx4.SSx4.p1.1" class="ltx_p">WC models can also incorporate separately published models of individual pathways. Currently, there are several model repositories which contain numerous cell cycle, circadian rhythm, electrical signaling, signal transduction, and metabolism models (Table S4–S5). However, most pathways such as RNA degradation do not yet have genome-scale dynamical models, many reported models are not publicly available, and it is difficult to merge most published models because they often use different assumptions and representations.</p>
</div>
</section>
</section>
<section id="Sx5" class="ltx_section">
<h2 class="ltx_title ltx_font_bold ltx_title_section">Emerging methods and tools</h2>

<div id="Sx5.p1" class="ltx_para ltx_noindent">
<p id="Sx5.p1.1" class="ltx_p">Recent advances in data aggregation, model design, model representation, and simulation (Table S6) are also rapidly making WC modeling feasible. We expect that ongoing efforts to adapt and combine these advances will accelerate WC modeling <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib9" title="" class="ltx_ref">9</a>]</cite> (Figure <a href="#Sx5.F2" title="Figure 2 ‣ Emerging methods and tools ‣ Emerging whole-cell modeling principles and methods" class="ltx_ref"><span class="ltx_text ltx_ref_tag">2</span></a>). Here, we summarize the most important emerging methods and tools for WC modeling.</p>
</div>
<figure id="Sx5.F2" class="ltx_figure"><img src="/html/1710.02431/assets/x2.png" id="Sx5.F2.g1" class="ltx_graphics ltx_centering ltx_img_landscape" width="461" height="98" alt="Refer to caption">
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">Figure 2: </span> Emerging WC modeling methodology. (a) Data should be aggregated from thousands of publications, repositories, and prediction tools and organized into a PGDB. (b) Models should be designed, calibrated, and validated from PGDBs and described using rules. (c) Models should be simulated using parallel, network-free, multi-algorithmic simulators and their results should be stored in a database. (d) Simulation results should be visualized and analyzed. (e) Results should be validated by comparison to experimental measurements. Importantly, all of these steps should be collaborative.</figcaption>
</figure>
<section id="Sx5.SSx1" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Data aggregation and organization</h3>

<div id="Sx5.SSx1.p1" class="ltx_para ltx_noindent">
<p id="Sx5.SSx1.p1.1" class="ltx_p">For optimal accuracy and scope, WC modeling should be tightly coupled with targeted experimentation. Nevertheless, we believe that WC modeling currently can be most cost-effectively advanced by leveraging the extensive array of public data. To make this public data usable for modeling, researchers are developing automated methods for extracting data from publications <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib29" title="" class="ltx_ref">29</a>]</cite>, building central public repositories <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib30" title="" class="ltx_ref">30</a>]</cite>, and creating tools for programmatically accessing repositories <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib31" title="" class="ltx_ref">31</a>]</cite>. Pathway/genome database (PGDB) tools such as Pathway Tools <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib32" title="" class="ltx_ref">32</a>]</cite> are well-suited to organizing this data because they support structured representations of metabolites, DNA, RNA, proteins, and their interactions. However, they provide limited support for non-metabolic pathways and quantitative data. To overcome these limitations, we developed the WholeCellKB tool to organize data for WC modeling <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib33" title="" class="ltx_ref">33</a>]</cite>.</p>
</div>
</section>
<section id="Sx5.SSx2" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Scalable model design</h3>

<div id="Sx5.SSx2.p1" class="ltx_para ltx_noindent">
<p id="Sx5.SSx2.p1.1" class="ltx_p">Several new tools can help researchers develop large models. For example, the Cell Collective facilitates collaborative model design <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib34" title="" class="ltx_ref">34</a>]</cite>, MetaFlux facilitates the design of constraint-based models from PGDBs <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib35" title="" class="ltx_ref">35</a>]</cite>, PySB facilitates programmatic model construction <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib36" title="" class="ltx_ref">36</a>]</cite>, SEEK facilitates model design from data tables <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib37" title="" class="ltx_ref">37</a>]</cite>, and Virtual Cell facilitates model design from KEGG and SABIO-RK <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib38" title="" class="ltx_ref">38</a>]</cite>.</p>
</div>
</section>
<section id="Sx5.SSx3" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Model languages</h3>

<div id="Sx5.SSx3.p1" class="ltx_para ltx_noindent">
<p id="Sx5.SSx3.p1.1" class="ltx_p">Researchers have developed several languages for representing biochemical models. SBML can represent several types of models including flux balance analysis models, deterministic dynamical models, and stochastic dynamical models <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib39" title="" class="ltx_ref">39</a>]</cite>. Rule-based languages such as BioNetGen can efficiently describe the combinatorial complexity of protein-protein interactions <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib40" title="" class="ltx_ref">40</a>]</cite>.</p>
</div>
</section>
<section id="Sx5.SSx4" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Simulation</h3>

<div id="Sx5.SSx4.p1" class="ltx_para ltx_noindent">
<p id="Sx5.SSx4.p1.1" class="ltx_p">Numerous tools can simulate biomodels. For example, COPASI <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib41" title="" class="ltx_ref">41</a>]</cite> and Virtual Cell <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib38" title="" class="ltx_ref">38</a>]</cite> support deterministic, stochastic, hybrid deterministic/stochastic, network-free, and spatial simulation; COBRApy supports constraint-based simulation <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib42" title="" class="ltx_ref">42</a>]</cite>; and E-Cell supports multi-algorithmic simulation <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib43" title="" class="ltx_ref">43</a>]</cite>.</p>
</div>
</section>
<section id="Sx5.SSx5" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Calibration</h3>

<div id="Sx5.SSx5.p1" class="ltx_para ltx_noindent">
<p id="Sx5.SSx5.p1.1" class="ltx_p">New tools such as saCeSS <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib44" title="" class="ltx_ref">44</a>]</cite> support distributed calibration of large biochemical models. In addition, aerospace and mechanical engineers have developed methods for using reduced surrogate models to efficiently calibrate large models <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib45" title="" class="ltx_ref">45</a>]</cite>.</p>
</div>
</section>
<section id="Sx5.SSx6" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Verification</h3>

<div id="Sx5.SSx6.p1" class="ltx_para ltx_noindent">
<p id="Sx5.SSx6.p1.1" class="ltx_p">Researchers have begun to adapt formal model checking techniques to biomodeling. For example, BioLab <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib46" title="" class="ltx_ref">46</a>]</cite> and PRISM <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib47" title="" class="ltx_ref">47</a>]</cite> can verify BioNetGen-encoded and SBML-encoded models, respectively.</p>
</div>
</section>
<section id="Sx5.SSx7" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Simulation results analysis</h3>

<div id="Sx5.SSx7.p1" class="ltx_para ltx_noindent">
<p id="Sx5.SSx7.p1.1" class="ltx_p">Tools such as COPASI <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib41" title="" class="ltx_ref">41</a>]</cite> and Virtual Cell <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib38" title="" class="ltx_ref">38</a>]</cite> can visualize simulation results. We have developed the WholeCellSimDB <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib48" title="" class="ltx_ref">48</a>]</cite> simulation results database to help researchers organize, search, and share WC simulation results. We have also developed the WholeCellViz <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib49" title="" class="ltx_ref">49</a>]</cite> simulation results dashboard to help researchers visualize WC simulation results in their biological context.</p>
</div>
</section>
</section>
<section id="Sx6" class="ltx_section">
<h2 class="ltx_title ltx_font_bold ltx_title_section">Technological challenges</h2>

<div id="Sx6.p1" class="ltx_para ltx_noindent">
<p id="Sx6.p1.1" class="ltx_p">Beyond these emerging tools, several technological advances are needed to enable WC models. Here, we summarize the most critically needed technologies.</p>
</div>
<section id="Sx6.SSx1" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Experimental measurement</h3>

<div id="Sx6.SSx1.p1" class="ltx_para ltx_noindent">
<p id="Sx6.SSx1.p1.1" class="ltx_p">While substantial data about cellular populations already exists, additional data would enable better models. In particular, we need metabolome-wide and proteome-wide measurement technologies that can quantitate the dynamics and single-cell variation of each metabolite and protein. Additionally, we need technologies that can measure kinetic parameters at the interactome scale and technologies that can measure cellular phenotypes across multiple genetic and environmental conditions. Furthermore, to enable WC models of a broad range of organisms, we also need technologies that can measure non-model organisms, including small, slow-growing, motile, and unculturable organisms.</p>
</div>
</section>
<section id="Sx6.SSx2" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Prediction tools</h3>

<div id="Sx6.SSx2.p1" class="ltx_para ltx_noindent">
<p id="Sx6.SSx2.p1.1" class="ltx_p">While existing tools can predict many properties of metabolites, DNA, RNA, and proteins, additional tools are needed to accurately predict the molecular effects of insertions, deletions, and structural variants. Such tools would help WC models design microbial genomes and predict the phenotypes of individual patients.</p>
</div>
</section>
<section id="Sx6.SSx3" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Data aggregation</h3>

<div id="Sx6.SSx3.p1" class="ltx_para ltx_noindent">
<p id="Sx6.SSx3.p1.1" class="ltx_p">As described above, extensive data is now available for WC modeling. However, this data is scattered across many repositories and publications; spans a wide range of data types, organisms, and environments; is described using inconsistent identifiers and units; and often is not annotated or normalized. To make this data more usable for modeling, we are developing a framework for aggregating data from repositories; merging data from multiple species, environmental conditions, and experimental procedures; standardizing data to common units; and identifying the most relevant data for a model.</p>
</div>
</section>
<section id="Sx6.SSx4" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Scalable, data-driven model design</h3>

<div id="Sx6.SSx4.p1" class="ltx_para ltx_noindent">
<p id="Sx6.SSx4.p1.1" class="ltx_p">To scale WC modeling, we need tools for collaboratively building large models directly from experimental data, recording how data is used to build models, and identifying gaps and inconsistencies in models. As described above, several tools support each of these functions. To accelerate WC modeling, the field must develop an extensible platform that supports all of these functions at the scale required for WC modeling.</p>
</div>
</section>
<section id="Sx6.SSx5" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Rule-based model representation</h3>

<div id="Sx6.SSx5.p1" class="ltx_para ltx_noindent">
<p id="Sx6.SSx5.p1.1" class="ltx_p">Several languages can represent individual biological processes, but no existing language supports all of the biological processes that WC models must represent <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib50" title="" class="ltx_ref">50</a>, <a href="#bib.bib51" title="" class="ltx_ref">51</a>]</cite>. To overcome this limitation, we are developing a rule-based language that can represent each molecular species at multiple levels of granularity (for example, as a single species, as a set of sites, and as a sequence); the combinatorial complexity of each molecular species and interaction; composite, multi-algorithmic models; and the data used to build models.</p>
</div>
</section>
<section id="Sx6.SSx6" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Scalable multi-algorithmic simulation</h3>

<div id="Sx6.SSx6.p1" class="ltx_para ltx_noindent">
<p id="Sx6.SSx6.p1.1" class="ltx_p">Simulating WC models requires a simulator that supports both network-free interpretation of rule-based model descriptions and multi-algorithmic co-simulation of submodels that are described using different simulation algorithms. However, no existing simulator supports both network-free and multi-algorithmic simulation. To scalably simulate WC models, we are using Rete algorithms and parallel discrete event simulation to develop a parallel, network-free, multi-algorithmic simulator <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib9" title="" class="ltx_ref">9</a>]</cite>.</p>
</div>
</section>
<section id="Sx6.SSx7" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Calibration and verification</h3>

<div id="Sx6.SSx7.p1" class="ltx_para ltx_noindent">
<p id="Sx6.SSx7.p1.1" class="ltx_p">Scalable tools are needed to calibrate and verify WC models. Although we and others have begun to explore surrogate strategies for efficiently calibrating and validating WC models <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib52" title="" class="ltx_ref">52</a>]</cite>, further work is needed to formalize these methods.</p>
</div>
</section>
<section id="Sx6.SSx8" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Simulation analysis</h3>

<div id="Sx6.SSx8.p1" class="ltx_para ltx_noindent">
<p id="Sx6.SSx8.p1.1" class="ltx_p">We and others have developed tools for organizing and visualizing simulation results, but they provided limited support for large datasets or custom visualizations such as pathway maps. To visualize WC simulation results, researchers should use distributed database and data processing technologies to search and reduce simulation results, standard visualization grammars to enable flexible and custom visualizations, and high-performance visualization toolkits to handle terabyte-scale simulation results.</p>
</div>
</section>
<section id="Sx6.SSx9" class="ltx_subsection">
<h3 class="ltx_title ltx_font_italic ltx_title_subsection">Collaboration</h3>

<div id="Sx6.SSx9.p1" class="ltx_para ltx_noindent">
<p id="Sx6.SSx9.p1.1" class="ltx_p">Ultimately, achieving WC models will require extensive teamwork. To facilitate collaboration, the field must develop collaborative model design tools, version control systems for models, standards for annotating and verifying submodels, and protocols for merging separately developed submodels.</p>
</div>
</section>
</section>
<section id="Sx7" class="ltx_section">
<h2 class="ltx_title ltx_font_bold ltx_title_section">Conclusion</h2>

<div id="Sx7.p1" class="ltx_para ltx_noindent">
<p id="Sx7.p1.1" class="ltx_p">WC models have great potential to advance bioscience, bioengineering, and medicine. However, significant challenges remain to achieve WC models. To advance WC modeling, we have proposed how WC models should represent cells and the phenotypes that WC models should predict, and summarized the best emerging methods and resources. We have also outlined several technological solutions to the most immediate WC modeling challenges. Specifically, we must develop new tools for scalably and collaboratively designing, simulating, calibrating, validating and analyzing models. We must also develop new methods for measuring the dynamics and single-cell variation of the metabolome and proteome and for measuring kinetic parameters at the interactome scale. Despite these challenges, we and others are building the first WC models, developing the first WC modeling tools, and beginning to form a WC modeling community <cite class="ltx_cite ltx_citemacro_cite">[<a href="#bib.bib52" title="" class="ltx_ref">52</a>, <a href="#bib.bib50" title="" class="ltx_ref">50</a>]</cite>. We anticipate that these efforts will enable comprehensive models of cells.</p>
</div>
</section>
<section id="Sx8" class="ltx_section">
<h2 class="ltx_title ltx_font_bold ltx_title_section">Acknowledgements</h2>

<div id="Sx8.p1" class="ltx_para ltx_noindent">
<p id="Sx8.p1.1" class="ltx_p">We thank Saahith Pochiraju for critical feedback. This work was supported by a National Institute of Health MIRA award [grant number 1 R35 GM119771-01]; a National Science Foundation INSPIRE award [grant number 1649014]; and the National Science Foundation / ERASynBio [grant numbers 1548123, 335672].</p>
</div>
</section>
<section id="Sx9" class="ltx_section">
<h2 class="ltx_title ltx_font_bold ltx_title_section">References and recommended reading</h2>

<div id="Sx9.p1" class="ltx_para ltx_noindent">
<p id="Sx9.p1.1" class="ltx_p">Papers of particular interest, published within the period of review, have been highlighted as:</p>
<dl id="Sx9.I1" class="ltx_description">
<dt id="Sx9.I1.ix1" class="ltx_item"><span class="ltx_tag ltx_tag_item"><math id="Sx9.I1.ix1.1.1.m1.1" class="ltx_Math" alttext="\bullet" display="inline"><semantics id="Sx9.I1.ix1.1.1.m1.1b"><mo id="Sx9.I1.ix1.1.1.m1.1.1" xref="Sx9.I1.ix1.1.1.m1.1.1.cmml">∙</mo><annotation-xml encoding="MathML-Content" id="Sx9.I1.ix1.1.1.m1.1c"><ci id="Sx9.I1.ix1.1.1.m1.1.1.cmml" xref="Sx9.I1.ix1.1.1.m1.1.1">∙</ci></annotation-xml><annotation encoding="application/x-tex" id="Sx9.I1.ix1.1.1.m1.1d">\bullet</annotation></semantics></math></span></dt>
<dd class="ltx_item">
<div id="Sx9.I1.ix1.p1" class="ltx_para">
<p id="Sx9.I1.ix1.p1.1" class="ltx_p">of special interest</p>
</div>
</dd>
<dt id="Sx9.I1.ix2" class="ltx_item"><span class="ltx_tag ltx_tag_item"><math id="Sx9.I1.ix2.1.1.m1.2" class="ltx_Math" alttext="\bullet\bullet" display="inline"><semantics id="Sx9.I1.ix2.1.1.m1.2b"><mrow id="Sx9.I1.ix2.1.1.m1.2.3.2" xref="Sx9.I1.ix2.1.1.m1.2.3.1.cmml"><mo id="Sx9.I1.ix2.1.1.m1.1.1" xref="Sx9.I1.ix2.1.1.m1.1.1.cmml">∙</mo><mo lspace="0em" id="Sx9.I1.ix2.1.1.m1.2.3.2.1" xref="Sx9.I1.ix2.1.1.m1.2.3.1.cmml">⁣</mo><mo id="Sx9.I1.ix2.1.1.m1.2.2" xref="Sx9.I1.ix2.1.1.m1.2.2.cmml">∙</mo></mrow><annotation-xml encoding="MathML-Content" id="Sx9.I1.ix2.1.1.m1.2c"><list id="Sx9.I1.ix2.1.1.m1.2.3.1.cmml" xref="Sx9.I1.ix2.1.1.m1.2.3.2"><ci id="Sx9.I1.ix2.1.1.m1.1.1.cmml" xref="Sx9.I1.ix2.1.1.m1.1.1">∙</ci><ci id="Sx9.I1.ix2.1.1.m1.2.2.cmml" xref="Sx9.I1.ix2.1.1.m1.2.2">∙</ci></list></annotation-xml><annotation encoding="application/x-tex" id="Sx9.I1.ix2.1.1.m1.2d">\bullet\bullet</annotation></semantics></math></span></dt>
<dd class="ltx_item">
<div id="Sx9.I1.ix2.p1" class="ltx_para ltx_noindent">
<p id="Sx9.I1.ix2.p1.1" class="ltx_p">of outstanding interest</p>
</div>
</dd>
</dl>
</div>
</section>
<section id="bib" class="ltx_bibliography">
<h2 class="ltx_title ltx_title_bibliography">References</h2>

<ul class="ltx_biblist">
<li id="bib.bib1" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[1]</span>
<span class="ltx_bibblock">
J. R. Karr, K. Takahashi, A. Funahashi, The principles of whole-cell modeling,
Curr Opin Microbiol 27 (2015) 18–24, <math id="bib.bib1.1.m1.2" class="ltx_Math" alttext="\bullet\bullet" display="inline"><semantics id="bib.bib1.1.m1.2a"><mrow id="bib.bib1.1.m1.2.3.2" xref="bib.bib1.1.m1.2.3.1.cmml"><mo id="bib.bib1.1.m1.1.1" xref="bib.bib1.1.m1.1.1.cmml">∙</mo><mo lspace="0em" id="bib.bib1.1.m1.2.3.2.1" xref="bib.bib1.1.m1.2.3.1.cmml">⁣</mo><mo id="bib.bib1.1.m1.2.2" xref="bib.bib1.1.m1.2.2.cmml">∙</mo></mrow><annotation-xml encoding="MathML-Content" id="bib.bib1.1.m1.2b"><list id="bib.bib1.1.m1.2.3.1.cmml" xref="bib.bib1.1.m1.2.3.2"><ci id="bib.bib1.1.m1.1.1.cmml" xref="bib.bib1.1.m1.1.1">∙</ci><ci id="bib.bib1.1.m1.2.2.cmml" xref="bib.bib1.1.m1.2.2">∙</ci></list></annotation-xml><annotation encoding="application/x-tex" id="bib.bib1.1.m1.2c">\bullet\bullet</annotation></semantics></math> Describes the
principles of WC modeling including mechanistic representation of each gene,
molecular species, and molecular interaction over the lifecycle of an
organism.

</span>
</li>
<li id="bib.bib2" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[2]</span>
<span class="ltx_bibblock">
M. Tomita, Whole-cell simulation: a grand challenge of the 21st century, Trends
Biotechnol 19 (6) (2001) 205–210, <math id="bib.bib2.1.m1.1" class="ltx_Math" alttext="\bullet" display="inline"><semantics id="bib.bib2.1.m1.1a"><mo id="bib.bib2.1.m1.1.1" xref="bib.bib2.1.m1.1.1.cmml">∙</mo><annotation-xml encoding="MathML-Content" id="bib.bib2.1.m1.1b"><ci id="bib.bib2.1.m1.1.1.cmml" xref="bib.bib2.1.m1.1.1">∙</ci></annotation-xml><annotation encoding="application/x-tex" id="bib.bib2.1.m1.1c">\bullet</annotation></semantics></math> Describes the need to develop WC
models to understand biology and personalize medicine.

</span>
</li>
<li id="bib.bib3" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[3]</span>
<span class="ltx_bibblock">
J. Carrera, M. W. Covert, Why build whole-cell models?, Trends Cell Biol
25 (12) (2015) 719–722, <math id="bib.bib3.1.m1.2" class="ltx_Math" alttext="\bullet\bullet" display="inline"><semantics id="bib.bib3.1.m1.2a"><mrow id="bib.bib3.1.m1.2.3.2" xref="bib.bib3.1.m1.2.3.1.cmml"><mo id="bib.bib3.1.m1.1.1" xref="bib.bib3.1.m1.1.1.cmml">∙</mo><mo lspace="0em" id="bib.bib3.1.m1.2.3.2.1" xref="bib.bib3.1.m1.2.3.1.cmml">⁣</mo><mo id="bib.bib3.1.m1.2.2" xref="bib.bib3.1.m1.2.2.cmml">∙</mo></mrow><annotation-xml encoding="MathML-Content" id="bib.bib3.1.m1.2b"><list id="bib.bib3.1.m1.2.3.1.cmml" xref="bib.bib3.1.m1.2.3.2"><ci id="bib.bib3.1.m1.1.1.cmml" xref="bib.bib3.1.m1.1.1">∙</ci><ci id="bib.bib3.1.m1.2.2.cmml" xref="bib.bib3.1.m1.2.2">∙</ci></list></annotation-xml><annotation encoding="application/x-tex" id="bib.bib3.1.m1.2c">\bullet\bullet</annotation></semantics></math> Describes several potential uses of
WC models including data integration, knowledge evaluation, phenotype
prediction, hypothesis generation, and genome design.

</span>
</li>
<li id="bib.bib4" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[4]</span>
<span class="ltx_bibblock">
M. Tomita, K. Hashimoto, K. Takahashi, T. S. Shimizu, Y. Matsuzaki, F. Miyoshi,
K. Saito, S. Tanida, K. Yugi, J. C. Venter, et al., E-CELL: software
environment for whole-cell simulation., Bioinformatics 15 (1) (1999) 72–84,
<math id="bib.bib4.1.m1.1" class="ltx_Math" alttext="\bullet" display="inline"><semantics id="bib.bib4.1.m1.1a"><mo id="bib.bib4.1.m1.1.1" xref="bib.bib4.1.m1.1.1.cmml">∙</mo><annotation-xml encoding="MathML-Content" id="bib.bib4.1.m1.1b"><ci id="bib.bib4.1.m1.1.1.cmml" xref="bib.bib4.1.m1.1.1">∙</ci></annotation-xml><annotation encoding="application/x-tex" id="bib.bib4.1.m1.1c">\bullet</annotation></semantics></math> Describes an early effort to build models that represent each gene
product.

</span>
</li>
<li id="bib.bib5" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[5]</span>
<span class="ltx_bibblock">
J. Atlas, E. Nikolaev, S. Browning, M. Shuler, Incorporating genome-wide DNA
sequence information into a dynamic whole-cell model of Escherichia coli:
application to DNA replication, IET systems biology 2 (5) (2008) 369–382,
<math id="bib.bib5.1.m1.1" class="ltx_Math" alttext="\bullet" display="inline"><semantics id="bib.bib5.1.m1.1a"><mo id="bib.bib5.1.m1.1.1" xref="bib.bib5.1.m1.1.1.cmml">∙</mo><annotation-xml encoding="MathML-Content" id="bib.bib5.1.m1.1b"><ci id="bib.bib5.1.m1.1.1.cmml" xref="bib.bib5.1.m1.1.1">∙</ci></annotation-xml><annotation encoding="application/x-tex" id="bib.bib5.1.m1.1c">\bullet</annotation></semantics></math> Describes an early effort to build WC models.

</span>
</li>
<li id="bib.bib6" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[6]</span>
<span class="ltx_bibblock">
E. Roberts, A. Magis, J. O. Ortiz, W. Baumeister, Z. Luthey-Schulten, Noise
contributions in an inducible genetic switch: a whole-cell simulation study,
PLoS computational biology 7 (3) (2011) e1002010.

</span>
</li>
<li id="bib.bib7" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[7]</span>
<span class="ltx_bibblock">
J. R. Karr, J. C. Sanghvi, D. N. Macklin, M. V. Gutschow, J. M. Jacobs,
B. Bolival, N. Assad-Garcia, J. I. Glass, M. W. Covert, A whole-cell
computational model predicts phenotype from genotype, Cell 150 (2) (2012)
389–401, <math id="bib.bib7.1.m1.2" class="ltx_Math" alttext="\bullet\bullet" display="inline"><semantics id="bib.bib7.1.m1.2a"><mrow id="bib.bib7.1.m1.2.3.2" xref="bib.bib7.1.m1.2.3.1.cmml"><mo id="bib.bib7.1.m1.1.1" xref="bib.bib7.1.m1.1.1.cmml">∙</mo><mo lspace="0em" id="bib.bib7.1.m1.2.3.2.1" xref="bib.bib7.1.m1.2.3.1.cmml">⁣</mo><mo id="bib.bib7.1.m1.2.2" xref="bib.bib7.1.m1.2.2.cmml">∙</mo></mrow><annotation-xml encoding="MathML-Content" id="bib.bib7.1.m1.2b"><list id="bib.bib7.1.m1.2.3.1.cmml" xref="bib.bib7.1.m1.2.3.2"><ci id="bib.bib7.1.m1.1.1.cmml" xref="bib.bib7.1.m1.1.1">∙</ci><ci id="bib.bib7.1.m1.2.2.cmml" xref="bib.bib7.1.m1.2.2">∙</ci></list></annotation-xml><annotation encoding="application/x-tex" id="bib.bib7.1.m1.2c">\bullet\bullet</annotation></semantics></math> Describes a novel approach for combining
heterogeneous data and models, and the first model that represents the
function of each characterized gene of an organism.

</span>
</li>
<li id="bib.bib8" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[8]</span>
<span class="ltx_bibblock">
A. Bordbar, D. McCloskey, D. C. Zielinski, N. Sonnenschein, N. Jamshidi, B. O.
Palsson, Personalized whole-cell kinetic models of metabolism for discovery
in genomics and pharmacodynamics, Cell Syst 1 (4) (2015) 283–292, <math id="bib.bib8.1.m1.1" class="ltx_Math" alttext="\bullet" display="inline"><semantics id="bib.bib8.1.m1.1a"><mo id="bib.bib8.1.m1.1.1" xref="bib.bib8.1.m1.1.1.cmml">∙</mo><annotation-xml encoding="MathML-Content" id="bib.bib8.1.m1.1b"><ci id="bib.bib8.1.m1.1.1.cmml" xref="bib.bib8.1.m1.1.1">∙</ci></annotation-xml><annotation encoding="application/x-tex" id="bib.bib8.1.m1.1c">\bullet</annotation></semantics></math>
Describes the development of genome-scale kinetic models of cellular
metabolism.

</span>
</li>
<li id="bib.bib9" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[9]</span>
<span class="ltx_bibblock">
A. P. Goldberg, Y. H. Chew, J. R. Karr, Toward scalable whole-cell modeling of
human cells, in: Proc 2016 Annu ACM Conf SIGSIM Princip Adv Discrete Simul,
ACM, 2016, pp. 259–262, <math id="bib.bib9.1.m1.1" class="ltx_Math" alttext="\bullet" display="inline"><semantics id="bib.bib9.1.m1.1a"><mo id="bib.bib9.1.m1.1.1" xref="bib.bib9.1.m1.1.1.cmml">∙</mo><annotation-xml encoding="MathML-Content" id="bib.bib9.1.m1.1b"><ci id="bib.bib9.1.m1.1.1.cmml" xref="bib.bib9.1.m1.1.1">∙</ci></annotation-xml><annotation encoding="application/x-tex" id="bib.bib9.1.m1.1c">\bullet</annotation></semantics></math> Proposes a parallel algorithm for
simulating multi-algorithmic WC models.

</span>
</li>
<li id="bib.bib10" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[10]</span>
<span class="ltx_bibblock">
B. Szigeti, Y. D. Roth, J. A. P. Sekar, A. P. Goldberg, S. Pochiraju, J. R.
Karr, A blueprint for human whole-cell modeling, Curr Opin Syst
Biol<math id="bib.bib10.1.m1.2" class="ltx_Math" alttext="\bullet\bullet" display="inline"><semantics id="bib.bib10.1.m1.2a"><mrow id="bib.bib10.1.m1.2.3.2" xref="bib.bib10.1.m1.2.3.1.cmml"><mo id="bib.bib10.1.m1.1.1" xref="bib.bib10.1.m1.1.1.cmml">∙</mo><mo lspace="0em" id="bib.bib10.1.m1.2.3.2.1" xref="bib.bib10.1.m1.2.3.1.cmml">⁣</mo><mo id="bib.bib10.1.m1.2.2" xref="bib.bib10.1.m1.2.2.cmml">∙</mo></mrow><annotation-xml encoding="MathML-Content" id="bib.bib10.1.m1.2b"><list id="bib.bib10.1.m1.2.3.1.cmml" xref="bib.bib10.1.m1.2.3.2"><ci id="bib.bib10.1.m1.1.1.cmml" xref="bib.bib10.1.m1.1.1">∙</ci><ci id="bib.bib10.1.m1.2.2.cmml" xref="bib.bib10.1.m1.2.2">∙</ci></list></annotation-xml><annotation encoding="application/x-tex" id="bib.bib10.1.m1.2c">\bullet\bullet</annotation></semantics></math> Describes the major bottlenecks to WC modeling and
proposes a community project to overcome these bottlenecks.

</span>
</li>
<li id="bib.bib11" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[11]</span>
<span class="ltx_bibblock">
D. N. Macklin, N. A. Ruggero, M. W. Covert, The future of whole-cell modeling,
Curr Opin Biotechnol 28 (2014) 111–115, <math id="bib.bib11.1.m1.2" class="ltx_Math" alttext="\bullet\bullet" display="inline"><semantics id="bib.bib11.1.m1.2a"><mrow id="bib.bib11.1.m1.2.3.2" xref="bib.bib11.1.m1.2.3.1.cmml"><mo id="bib.bib11.1.m1.1.1" xref="bib.bib11.1.m1.1.1.cmml">∙</mo><mo lspace="0em" id="bib.bib11.1.m1.2.3.2.1" xref="bib.bib11.1.m1.2.3.1.cmml">⁣</mo><mo id="bib.bib11.1.m1.2.2" xref="bib.bib11.1.m1.2.2.cmml">∙</mo></mrow><annotation-xml encoding="MathML-Content" id="bib.bib11.1.m1.2b"><list id="bib.bib11.1.m1.2.3.1.cmml" xref="bib.bib11.1.m1.2.3.2"><ci id="bib.bib11.1.m1.1.1.cmml" xref="bib.bib11.1.m1.1.1">∙</ci><ci id="bib.bib11.1.m1.2.2.cmml" xref="bib.bib11.1.m1.2.2">∙</ci></list></annotation-xml><annotation encoding="application/x-tex" id="bib.bib11.1.m1.2c">\bullet\bullet</annotation></semantics></math> Summarizes the
challenges to achieve WC models including aggregating the data needed to
build WC models; merging pathway submodels; simulating, calibrating, and
validating large models; and coordinating large modeling teams.

</span>
</li>
<li id="bib.bib12" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[12]</span>
<span class="ltx_bibblock">
I. C. Macaulay, T. Voet, Single cell genomics: advances and future
perspectives, PLoS Genet 10 (1) (2014) e1004126.

</span>
</li>
<li id="bib.bib13" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[13]</span>
<span class="ltx_bibblock">
A. M. Altelaar, J. Munoz, A. J. Heck, Next-generation proteomics: towards an
integrative view of proteome dynamics, Nat Rev Genet 14 (1) (2013) 35.

</span>
</li>
<li id="bib.bib14" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[14]</span>
<span class="ltx_bibblock">
T. Fuhrer, N. Zamboni, High-throughput discovery metabolomics, Curr Opinion
Biotechnol 31 (2015) 73–78.

</span>
</li>
<li id="bib.bib15" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[15]</span>
<span class="ltx_bibblock">
P. W. Laird, Principles and challenges of genome-wide DNA methylation
analysis, Nat Rev Genetics 11 (3) (2010) 191.

</span>
</li>
<li id="bib.bib16" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[16]</span>
<span class="ltx_bibblock">
J. Dekker, M. A. Marti-Renom, L. A. Mirny, Exploring the three-dimensional
organization of genomes: interpreting chromatin interaction data, Nat Rev
Genet 14 (6) (2013) 390.

</span>
</li>
<li id="bib.bib17" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[17]</span>
<span class="ltx_bibblock">
P. J. Park, ChIP-seq: advantages and challenges of a maturing technology, Nat
Rev Genet 10 (10) (2009) 669.

</span>
</li>
<li id="bib.bib18" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[18]</span>
<span class="ltx_bibblock">
J. H. Lee, E. R. Daugharthy, J. Scheiman, R. Kalhor, J. L. Yang, T. C.
Ferrante, R. Terry, S. S. Jeanty, C. Li, R. Amamoto, et al., Highly
multiplexed subcellular RNA sequencing in situ, Science 343 (6177) (2014)
1360–1363.

</span>
</li>
<li id="bib.bib19" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[19]</span>
<span class="ltx_bibblock">
A.-E. Saliba, A. J. Westermann, S. A. Gorski, J. Vogel, Single-cell RNA-seq:
advances and future challenges, Nucleic Acids Res 42 (14) (2014) 8845–8860.

</span>
</li>
<li id="bib.bib20" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[20]</span>
<span class="ltx_bibblock">
S. C. Bendall, G. P. Nolan, M. Roederer, P. K. Chattopadhyay, A deep profiler’s
guide to cytometry, Trends Immunol 33 (7) (2012) 323–332.

</span>
</li>
<li id="bib.bib21" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[21]</span>
<span class="ltx_bibblock">
T. U. Consortium, UniProt: the universal protein knowledgebase, Nucleic Acids
Res 45 (D1) (2017) D158–D169.

</span>
</li>
<li id="bib.bib22" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[22]</span>
<span class="ltx_bibblock">
R. Caspi, R. Billington, L. Ferrer, H. Foerster, C. A. Fulcher, I. M. Keseler,
A. Kothari, M. Krummenacker, M. Latendresse, L. A. Mueller, et al., The
MetaCyc database of metabolic pathways and enzymes and the BioCyc
collection of pathway/genome databases, Nucleic Acids Res 44 (D1) (2016)
D471–D480.

</span>
</li>
<li id="bib.bib23" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[23]</span>
<span class="ltx_bibblock">
T. Sajed, A. Marcu, M. Ramirez, A. Pon, A. C. Guo, C. Knox, M. Wilson, J. R.
Grant, Y. Djoumbou, D. S. Wishart, ECMDB 2.0: A richer resource for
understanding the biochemistry of E. coli, Nucleic Acids Res 44 (D1) (2016)
D495–D501.

</span>
</li>
<li id="bib.bib24" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[24]</span>
<span class="ltx_bibblock">
N. Kolesnikov, E. Hastings, M. Keays, O. Melnichuk, Y. A. Tang, E. Williams,
M. Dylag, N. Kurbatova, M. Brandizi, T. Burdett, et al., ArrayExpress
update–simplifying data submissions, Nucleic Acids Res 43 (D1) (2015)
D1113–D1116.

</span>
</li>
<li id="bib.bib25" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[25]</span>
<span class="ltx_bibblock">
M. Wang, C. J. Herrmann, M. Simonovic, D. Szklarczyk, C. Mering, Version 4.0 of
PaxDb: protein abundance data, integrated across model organisms, tissues,
and cell-lines, Proteomics 15 (18) (2015) 3163–3168.

</span>
</li>
<li id="bib.bib26" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[26]</span>
<span class="ltx_bibblock">
U. Wittig, R. Kania, M. Golebiewski, M. Rey, L. Shi, L. Jong, E. Algaa,
A. Weidemann, H. Sauer-Danzwith, S. Mir, et al., SABIO-RK–database for
biochemical reaction kinetics, Nucleic Acids Res 40 (D1) (2012) D790–D796.

</span>
</li>
<li id="bib.bib27" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[27]</span>
<span class="ltx_bibblock">
M. Y. Galperin, X. M. Fernández-Suárez, D. J. Rigden, The 24th annual
Nucleic Acids Research database issue: a look back and upcoming changes,
Nucleic Acids Res 45 (D1) (2017) D1–D11.

</span>
</li>
<li id="bib.bib28" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[28]</span>
<span class="ltx_bibblock">
N. Y. Yu, J. R. Wagner, M. R. Laird, G. Melli, S. Rey, R. Lo, P. Dao, S. C.
Sahinalp, M. Ester, L. J. Foster, et al., PSORTb 3.0: improved protein
subcellular localization prediction with refined localization subcategories
and predictive capabilities for all prokaryotes, Bioinformatics 26 (13)
(2010) 1608–1615.

</span>
</li>
<li id="bib.bib29" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[29]</span>
<span class="ltx_bibblock">
P. R. Cohen, DARPA’s Big Mechanism program, Phys Biol 12 (4) (2015) 045008.

</span>
</li>
<li id="bib.bib30" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[30]</span>
<span class="ltx_bibblock">
H. Pampel, P. Vierkant, F. Scholze, R. Bertelmann, M. Kindling, J. Klump, H.-J.
Goebelbecker, J. Gundlach, P. Schirmbacher, U. Dierolf, Making research data
repositories visible: The re3data.org Registry, PloS One 8 (11) (2013)
e78080.

</span>
</li>
<li id="bib.bib31" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[31]</span>
<span class="ltx_bibblock">
T. Cokelaer, D. Pultz, L. M. Harder, J. Serra-Musach, J. Saez-Rodriguez,
BioServices: a common Python package to access biological web services
programmatically, Bioinformatics 29 (24) (2013) 3241–3242.

</span>
</li>
<li id="bib.bib32" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[32]</span>
<span class="ltx_bibblock">
P. D. Karp, M. Latendresse, S. M. Paley, M. Krummenacker, Q. D. Ong,
R. Billington, A. Kothari, D. Weaver, T. Lee, P. Subhraveti, et al., Pathway
Tools version 19.0 update: software for pathway/genome informatics and
systems biology, Brief Bioinform 17 (5) (2016) 877–890.

</span>
</li>
<li id="bib.bib33" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[33]</span>
<span class="ltx_bibblock">
J. R. Karr, J. C. Sanghvi, D. N. Macklin, A. Arora, M. W. Covert,
WholeCellKB: model organism databases for comprehensive whole-cell models,
Nucleic Acids Res 41 (D1) (2013) D787–D792.

</span>
</li>
<li id="bib.bib34" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[34]</span>
<span class="ltx_bibblock">
T. Helikar, B. Kowal, J. Rogers, A cell simulator platform: the Cell
Collective, Clin Pharmacol Ther 93 (5) (2013) 393–395.

</span>
</li>
<li id="bib.bib35" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[35]</span>
<span class="ltx_bibblock">
M. Latendresse, M. Krummenacker, M. Trupp, P. D. Karp, Construction and
completion of flux balance models from pathway databases, Bioinformatics
28 (3) (2012) 388–396.

</span>
</li>
<li id="bib.bib36" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[36]</span>
<span class="ltx_bibblock">
C. F. Lopez, J. L. Muhlich, J. A. Bachman, P. K. Sorger, Programming biological
models in Python using PySB, Mol Syst Biol 9 (1) (2013) 646.

</span>
</li>
<li id="bib.bib37" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[37]</span>
<span class="ltx_bibblock">
K. Wolstencroft, S. Owen, O. Krebs, Q. Nguyen, N. J. Stanford, M. Golebiewski,
A. Weidemann, M. Bittkowski, L. An, D. Shockley, et al., SEEK: a systems
biology data and model management platform, BMC Syst Biol 9 (1) (2015) 33.

</span>
</li>
<li id="bib.bib38" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[38]</span>
<span class="ltx_bibblock">
D. C. Resasco, F. Gao, F. Morgan, I. L. Novak, J. C. Schaff, B. M. Slepchenko,
Virtual Cell: computational tools for modeling in cell biology, Wiley
Interdiscip Rev Syst Biol Med 4 (2) (2012) 129–140.

</span>
</li>
<li id="bib.bib39" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[39]</span>
<span class="ltx_bibblock">
M. Hucka, A. Finney, H. M. Sauro, H. Bolouri, J. C. Doyle, H. Kitano, A. P.
Arkin, B. J. Bornstein, D. Bray, A. Cornish-Bowden, et al., The systems
biology markup language (SBML): a medium for representation and exchange of
biochemical network models, Bioinformatics 19 (4) (2003) 524–531.

</span>
</li>
<li id="bib.bib40" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[40]</span>
<span class="ltx_bibblock">
L. A. Harris, J. S. Hogg, J.-J. Tapia, J. A. Sekar, S. Gupta, I. Korsunsky,
A. Arora, D. Barua, R. P. Sheehan, J. R. Faeder, BioNetGen 2.2: advances in
rule-based modeling, Bioinformatics 32 (21) (2016) 3366–3368.

</span>
</li>
<li id="bib.bib41" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[41]</span>
<span class="ltx_bibblock">
P. Mendes, S. Hoops, S. Sahle, R. Gauges, J. Dada, U. Kummer, Computational
modeling of biochemical networks using COPASI, Methods Mol Biol 500 (2009)
17–59.

</span>
</li>
<li id="bib.bib42" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[42]</span>
<span class="ltx_bibblock">
A. Ebrahim, J. A. Lerman, B. O. Palsson, D. R. Hyduke, Cobrapy:
Constraints-based reconstruction and analysis for python, BMC systems biology
7 (1) (2013) 74.

</span>
</li>
<li id="bib.bib43" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[43]</span>
<span class="ltx_bibblock">
P. K. Dhar, K. Takahashi, Y. Nakayama, M. Tomita, E-Cell: Computer simulation
of the cell, Rev Cell Biol Mol Med<math id="bib.bib43.1.m1.1" class="ltx_Math" alttext="\bullet" display="inline"><semantics id="bib.bib43.1.m1.1a"><mo id="bib.bib43.1.m1.1.1" xref="bib.bib43.1.m1.1.1.cmml">∙</mo><annotation-xml encoding="MathML-Content" id="bib.bib43.1.m1.1b"><ci id="bib.bib43.1.m1.1.1.cmml" xref="bib.bib43.1.m1.1.1">∙</ci></annotation-xml><annotation encoding="application/x-tex" id="bib.bib43.1.m1.1c">\bullet</annotation></semantics></math> Describes a multi-algorithmic
simulator.

</span>
</li>
<li id="bib.bib44" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[44]</span>
<span class="ltx_bibblock">
D. R. Penas, P. González, J. A. Egea, R. Doallo, J. R. Banga, Parameter
estimation in large-scale systems biology models: a parallel and
self-adaptive cooperative strategy, BMC Bioinformatics 18 (1) (2017) 52.

</span>
</li>
<li id="bib.bib45" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[45]</span>
<span class="ltx_bibblock">
A. I. Forrester, A. J. Keane, Recent advances in surrogate-based optimization,
Progress in Aerospace Sciences 45 (1) (2009) 50–79.

</span>
</li>
<li id="bib.bib46" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[46]</span>
<span class="ltx_bibblock">
E. M. Clarke, J. R. Faeder, C. J. Langmead, L. A. Harris, S. K. Jha, A. Legay,
Statistical model checking in BioLab: Applications to the automated
analysis of T-cell receptor signaling pathway, in: Int Conf Comput Meth
Syst Biol, Springer, 2008, pp. 231–250.

</span>
</li>
<li id="bib.bib47" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[47]</span>
<span class="ltx_bibblock">
M. Kwiatkowska, G. Norman, D. Parker, PRISM 4.0: Verification of
probabilistic real-time systems, in: Computer Aided Verification, Springer,
2011, pp. 585–591.

</span>
</li>
<li id="bib.bib48" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[48]</span>
<span class="ltx_bibblock">
J. R. Karr, N. C. Phillips, M. W. Covert, WholeCellSimDB: a hybrid
relational/HDF database for whole-cell model predictions, Database 2014.

</span>
</li>
<li id="bib.bib49" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[49]</span>
<span class="ltx_bibblock">
R. Lee, J. R. Karr, M. W. Covert, WholeCellViz: data visualization for
whole-cell models, BMC Bioinformatics 14 (1) (2013) 253.

</span>
</li>
<li id="bib.bib50" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[50]</span>
<span class="ltx_bibblock">
D. Waltemath, J. R. Karr, F. T. Bergmann, V. Chelliah, M. Hucka, M. Krantz,
W. Liebermeister, P. Mendes, C. J. Myers, P. Pir, et al., Toward community
standards and software for whole-cell modeling, IEEE Trans Biomed Eng 63 (10)
(2016) 2007–2014.

</span>
</li>
<li id="bib.bib51" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[51]</span>
<span class="ltx_bibblock">
J. K. Medley, A. P. Goldberg, J. R. Karr, Guidelines for reproducibly building
and simulating systems biology models, IEEE Trans Biomed Eng 63 (10) (2016)
2015–2020.

</span>
</li>
<li id="bib.bib52" class="ltx_bibitem">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">[52]</span>
<span class="ltx_bibblock">
J. R. Karr, A. H. Williams, J. D. Zucker, A. Raue, B. Steiert, J. Timmer,
C. Kreutz, S. Wilkinson, B. A. Allgood, B. M. Bot, et al., Summary of the
DREAM8 parameter estimation challenge: toward parameter identification for
whole-cell models, PLoS Comput Biol 11 (5) (2015) e1004096.

</span>
</li>
</ul>
</section>
</article>
</div>
<div class="ar5iv-footer"><a href="/html/1710.02430" class="ar5iv-nav-button ar5iv-nav-button-prev">◄</a>
    <a class="ar5iv-home-button" href="/"><img height="40" alt="ar5iv homepage" src="/assets/ar5iv.png"></a>
    <a href="/feeling_lucky" class="ar5iv-text-button">Feeling<br>lucky?</a>
    <a href="/land_of_honey_and_milk" rel="nofollow" aria-hidden="true" tabindex="-1"></a>
    <a href="/log/1710.02431" class="ar5iv-text-button ar5iv-severity-warning">Conversion<br>report</a>
    <a class="ar5iv-text-button" target="_blank" href="https://github.com/dginev/ar5iv/issues/new?template=improve-article--arxiv-id-.md&title=Improve+article+1710.02431">Report<br>an issue</a>
    <a href="https://arxiv.org/abs/1710.02431" class="ar5iv-text-button arxiv-ui-theme">View&nbsp;original<br>on&nbsp;arXiv</a><a href="/html/1710.02432" class="ar5iv-nav-button ar5iv-nav-button-next">►</a>
</div><footer class="ltx_page_footer">
<a class="ar5iv-toggle-color-scheme" href="javascript:toggleColorScheme()" title="Toggle ar5iv color scheme"><span class="color-scheme-icon"></span></a>
<a class="ar5iv-footer-button" href="https://arxiv.org/help/license" target="_blank">Copyright</a>
<a class="ar5iv-footer-button" href="https://arxiv.org/help/policies/privacy_policy" target="_blank">Privacy Policy</a>

<div class="ltx_page_logo">Generated  on Sun Mar  3 03:22:41 2024 by <a target="_blank" href="http://dlmf.nist.gov/LaTeXML/" class="ltx_LaTeXML_logo"><span style="letter-spacing:-0.2em; margin-right:0.1em;">L<span class="ltx_font_smallcaps" style="position:relative; bottom:2.2pt;">a</span>T<span class="ltx_font_smallcaps" style="font-size:120%;position:relative; bottom:-0.2ex;">e</span></span><span style="font-size:90%; position:relative; bottom:-0.2ex;">XML</span><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAsAAAAOCAYAAAD5YeaVAAAAAXNSR0IArs4c6QAAAAZiS0dEAP8A/wD/oL2nkwAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9wKExQZLWTEaOUAAAAddEVYdENvbW1lbnQAQ3JlYXRlZCB3aXRoIFRoZSBHSU1Q72QlbgAAAdpJREFUKM9tkL+L2nAARz9fPZNCKFapUn8kyI0e4iRHSR1Kb8ng0lJw6FYHFwv2LwhOpcWxTjeUunYqOmqd6hEoRDhtDWdA8ApRYsSUCDHNt5ul13vz4w0vWCgUnnEc975arX6ORqN3VqtVZbfbTQC4uEHANM3jSqXymFI6yWazP2KxWAXAL9zCUa1Wy2tXVxheKA9YNoR8Pt+aTqe4FVVVvz05O6MBhqUIBGk8Hn8HAOVy+T+XLJfLS4ZhTiRJgqIoVBRFIoric47jPnmeB1mW/9rr9ZpSSn3Lsmir1fJZlqWlUonKsvwWwD8ymc/nXwVBeLjf7xEKhdBut9Hr9WgmkyGEkJwsy5eHG5vN5g0AKIoCAEgkEkin0wQAfN9/cXPdheu6P33fBwB4ngcAcByHJpPJl+fn54mD3Gg0NrquXxeLRQAAwzAYj8cwTZPwPH9/sVg8PXweDAauqqr2cDjEer1GJBLBZDJBs9mE4zjwfZ85lAGg2+06hmGgXq+j3+/DsixYlgVN03a9Xu8jgCNCyIegIAgx13Vfd7vdu+FweG8YRkjXdWy329+dTgeSJD3ieZ7RNO0VAXAPwDEAO5VKndi2fWrb9jWl9Esul6PZbDY9Go1OZ7PZ9z/lyuD3OozU2wAAAABJRU5ErkJggg==" alt="Mascot Sammy"></a>
</div></footer>
</div>

    <script>
      var canMathML = typeof(MathMLElement) == "function";
      if (!canMathML) {
        var body = document.querySelector("body");
        body.firstElementChild.setAttribute('style', 'opacity: 0;');
        var loading = document.createElement("div");
        loading.setAttribute("id", "mathjax-loading-spinner");
        var message = document.createElement("div");
        message.setAttribute("id", "mathjax-loading-message");
        message.innerText = "Typesetting Equations...";
        body.prepend(loading);
        body.prepend(message);

        var el = document.createElement("script");
        el.src = "https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js";
        document.querySelector("head").appendChild(el);

        window.MathJax = {
          startup: {
            pageReady: () => {
              return MathJax.startup.defaultPageReady().then(() => {
                body.removeChild(loading);
                body.removeChild(message);
                body.firstElementChild.removeAttribute('style');
              }); } } };
      }
    </script>
    <script>
    // Auxiliary function, building the preview feature when
    // an inline citation is clicked
    function clicked_cite(e) {
      e.preventDefault();
      let cite = this.closest('.ltx_cite');
      let next = cite.nextSibling;
      if (next && next.nodeType == Node.ELEMENT_NODE && next.getAttribute('class') == "ar5iv-bibitem-preview") {
        next.remove();
        return; }
      // Before adding a preview modal,
      // cleanup older previews, in case they're still open
      document.querySelectorAll('span.ar5iv-bibitem-preview').forEach(function(node) {
        node.remove();
      })

      // Create the preview
      preview = document.createElement('span');
      preview.setAttribute('class','ar5iv-bibitem-preview');
      let target = document.getElementById(this.getAttribute('href').slice(1));
      target.childNodes.forEach(function (child) {
        preview.append(child.cloneNode(true));
      });
      let close_x = document.createElement('button');
      close_x.setAttribute("aria-label","Close modal for bibliography item preview");
      close_x.textContent = "×";
      close_x.setAttribute('class', 'ar5iv-button-close-preview');
      close_x.setAttribute('onclick','this.parentNode.remove()');
      preview.append(close_x);
      preview.querySelectorAll('.ltx_tag_bibitem').forEach(function(node) {
        node.remove();
      });
      cite.parentNode.insertBefore(preview, cite.nextSibling);
      return;
    }
    // Global Document initialization:
    // - assign the preview feature to all inline citation links
    document.querySelectorAll(".ltx_cite .ltx_ref").forEach(function (link) {
      link.addEventListener("click", clicked_cite);
    });
    </script>
    </body>
</html>
